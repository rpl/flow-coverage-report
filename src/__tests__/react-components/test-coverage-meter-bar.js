'use babel';

/* eslint-disable react/jsx-filename-extension, import/no-dynamic-require */

import React from 'react';
import renderer from 'react-test-renderer';

import {BASE_DIR} from './common';

const REACT_COMPONENT = `${BASE_DIR}/coverage-meter-bar`;

describe('<FlowCoverageMeterBar />', () => {
  test('percent <= threshold', () => {
    const FlowCoverageMeterBar = require(REACT_COMPONENT).default;
    const props = {
      percent: 20,
      threshold: 80
    };
    const tree = renderer.create(<FlowCoverageMeterBar {...props}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('uncoveredCount <= thresholdUncovered', () => {
    const FlowCoverageMeterBar = require(REACT_COMPONENT).default;
    const props = {
      uncoveredCount: 20,
      thresholdUncovered: 30
    };
    const tree = renderer.create(<FlowCoverageMeterBar {...props}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

test.skip('<FlowCoverageMeterBar /> with missing props');
