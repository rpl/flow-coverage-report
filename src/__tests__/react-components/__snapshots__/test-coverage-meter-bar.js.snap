// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlowCoverageMeterBar /> percent <= threshold 1`] = `
<div
  className="row red"
  style={
    Object {
      "height": 12,
      "padding": 0,
    }
  }
/>
`;

exports[`<FlowCoverageMeterBar /> uncoveredCount <= thresholdUncovered 1`] = `
<div
  className="row green"
  style={
    Object {
      "height": 12,
      "padding": 0,
    }
  }
/>
`;
